src/node/history.h

```diff
@@ -232,6 +242,76 @@ namespace ccf
         throw std::logic_error("Precondition to mt_retract_to violated");
       mt_retract_to(tree, index);
     }
+
+    std::
+      tuple<std::unique_ptr<hash_vec>, crypto::Sha256Hash, uint64_t, uint32_t>
```

Comment:
tuple could maybe be a struct for readability

Reply:
Yes, and also because the path requires custom management, done.